### PR TITLE
Fix _most_ NameErrors

### PR DESCRIPTION
**Pull request by jbrockmendel**
Fixes F821 warnings which point to lurking NameErrors.  In some cases these are easily fixed.  In others they are in rotted-to-pieces detritus, in which case any paths that would raise NameError were deleted.

There are still a handful left that I can neither fix nor delete.  Someone (@josef-pkt ) will need to decipher what these are supposed to do.

```
statsmodels/distributions/empirical_distribution.py:143:63: F821 undefined name 'ival'
statsmodels/sandbox/panel/panelmod.py:292:14: F821 undefined name 's'
statsmodels/stats/regularized_covariance.py:35:23: F821 undefined name 'ind'
statsmodels/stats/regularized_covariance.py:79:23: F821 undefined name 'ind'
```

### CODE REVIEW

**Comment by josef-pkt** [comment]
we are not deleting the currently unused code in the sandbox, at least not unitl having triaged the sandbox, some of those need refactoring and not deletion

**Comment by jbrockmendel** [comment]
> we are not deleting the currently unused code in the sandbox

The idea here was to illustrate the amount of entirely-unusable code.

>  at least not unitl having triaged the sandbox

Great, let's do it!

I'll revert the deletion portions of the PR (and keep the git history immaculate).  In the interim, can you please help with the 4 unhandled cases mentioned above?

File: tools/migrate_issues_gh.py [hunk 1]
```diff
@@ -51,6 +51,10 @@
 from launchpadlib.launchpad import (
```
Comment (josef-pkt) [inline tools/migrate_issues_gh.py]: we don't fix this module, it's here only for historical reasons (can be deleted in a "delete obsolete code" PR

**Comment by jbrockmendel** [comment]
superseded by #4859

**Pull request closed without merging**